pkg/sources/github/github.go

```diff
@@ -273,6 +273,65 @@ func (s *Source) Init(aCtx context.Context, name string, jobID, sourceID int64,
 	return nil
 }
 
+// Validate is used by enterprise CLI to validate the Github config file.
+func (s *Source) Validate(ctx context.Context) []error {
+	var (
+		errs     []error
+		ghClient *github.Client
+		err      error
+	)
+	apiEndpoint := s.conn.Endpoint
+
+	switch cred := s.conn.GetCredential().(type) {
+	case *sourcespb.GitHub_BasicAuth:
+		s.httpClient.Transport = &github.BasicAuthTransport{
+			Username: cred.BasicAuth.Username,
+			Password: cred.BasicAuth.Password,
+		}
+		ghClient, err = createGitHubClient(s.httpClient, apiEndpoint)
+		if err != nil {
+			errs = append(errs, err)
```

Comment:
The errors returned when creating the client /might/ not make it clear exactly where the failure is. For validate, it would be better to wrap with a bit of extra context. eg https://github.com/trufflesecurity/trufflehog/blob/948a94792b5a165fc4cb0a22412c96fc3513431e/pkg/sources/syslog/syslog.go#L80

Reply:
ah gotcha